src/main/java/seedu/address/model/income/Income.java

```diff
@@ -0,0 +1,76 @@
+package seedu.address.model.income;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDate;
+import java.util.Objects;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Represents an Income.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Income {
+    private final Name name;
+    private final Amount amount;
+    private final LocalDate date;
```

Comment:
Would it be better to use LocalDateTime instead of LocalDate since we eventually have to store time information?